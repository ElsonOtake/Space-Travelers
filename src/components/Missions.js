import React from 'react';

const Missions = () => (
  <h1>Missions</h1>
);

export default Missions;
