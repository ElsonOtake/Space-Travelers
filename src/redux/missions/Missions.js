import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { axiosGet } from '../api';

const FETCH = 'space-travelers/missions/FETCH';
const urlMissions = 'https://api.spacexdata.com/v3/missions';
const initialState = {
  loading: false,
};

const missionsReducer = (state = [], action = {}) => {
  switch (action.type) {
    case FETCH:
      return action.payload;
    default:
      return state;
  }
};

export const fetchMissions = createAsyncThunk(
  FETCH,
  async (_, { dispatch }) => {
    const res = await axiosGet(urlMissions)
      .then(
        (data) => dispatch({
          type: FETCH,
          payload: data,
        }),
      );
    return res;
  },
);

export const missionsSlice = createSlice({
  name: 'missionsSlice',
  initialState,
  extraReducers: {
    [fetchMissions.pending]: (state) => ({
      ...state,
      loading: true,
    }),
    [fetchMissions.fulfilled]: (state) => ({
      ...state,
      loading: false,
    }),
    [fetchMissions.rejected]: (state) => ({
      ...state,
      loading: false,
    }),
  },
});

export const missionsSliceReducer = missionsSlice.reducer;

export default missionsReducer;
