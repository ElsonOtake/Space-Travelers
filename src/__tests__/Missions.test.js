import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import store from '../redux/configureStore';

describe('Tests for the Missions component', () => {
  render(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );
  test('Check for the following test on screen', () => {
    expect(screen.getByText('Mission')).toBeInTheDocument();
    expect(screen.getByText('Description')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();

    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(screen.getByRole('table')).toBeVisible();
    expect(screen.getByRole('table')).toHaveClass('table-striped');
    expect(screen.getByRole('table')).toHaveClass('table-bordered');
    expect(screen.getByRole('table')).toHaveClass('table-hover');
    expect(screen.getByRole('table')).not.toHaveClass('striped');
    expect(screen.getByRole('table')).not.toHaveClass('bordered');
    expect(screen.getByRole('table')).not.toHaveClass('hover');

    expect(screen.getAllByRole('columnheader').length).toBe(4);
  });
  test('Check for the snapshot', () => {
    expect(screen.debug()).toMatchSnapshot();
  });
});
