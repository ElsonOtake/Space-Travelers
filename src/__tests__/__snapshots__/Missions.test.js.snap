// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Missions component Check for the snapshot 1`] = `undefined`;
